src/sentry/static/sentry/app/views/projects/redirectLegacyProjectRoute.jsx

```diff
@@ -0,0 +1,154 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import _ from 'lodash';
+
+import {t} from 'app/locale';
+import withApi from 'app/utils/withApi';
+import LoadingError from 'app/components/loadingError';
+import {analytics} from 'app/utils/analytics';
+import Alert from 'app/components/alert';
+import Redirect from 'app/utils/redirect';
+
+class ProjectDetailsInner extends React.Component {
+  static propTypes = {
+    api: PropTypes.object.isRequired,
+
+    orgId: PropTypes.string.isRequired,
+    projectSlug: PropTypes.string.isRequired,
+  };
+
+  state = {
+    loading: true,
+    error: null,
+    project: null,
+  };
+
+  componentDidMount() {
+    this.fetchData();
+  }
+
+  fetchData = async () => {
+    this.setState({
+      loading: true,
+      error: null,
+    });
+
+    const {orgId, projectSlug} = this.props;
+
+    try {
+      const project = await this.props.api.requestPromise(
+        `/projects/${orgId}/${projectSlug}/`
+      );
+
+      this.setState({
+        loading: false,
+        error: null,
+        project,
+      });
+    } catch (error) {
+      this.setState({
+        loading: false,
+        error,
+        project: null,
+      });
+    }
+  };
+
+  getProjectId = () => {
+    return this.state.project?.id;
+  };
+
+  hasProjectId = () => {
+    const projectID = this.getProjectId(this.state.project);
+    return _.isString(projectID) && projectID.length > 0;
+  };
+
+  render() {
+    const childrenProps = {
+      ...this.state,
+      projectId: this.getProjectId(),
+      hasProjectId: this.hasProjectId(),
+    };
+
+    return this.props.children(childrenProps);
+  }
+}
+
+export const ProjectDetails = withApi(ProjectDetailsInner);
+
+const redirectSentry9Project = generateRedirectRoute => {
+  class RedirectSentry9Project extends React.Component {
+    static propTypes = {
+      router: PropTypes.object.isRequired,
+
+      location: PropTypes.shape({
+        pathname: PropTypes.string.isRequired,
+        search: PropTypes.string.isRequired,
+      }).isRequired,
+
+      params: PropTypes.shape({
+        orgId: PropTypes.string.isRequired,
+        projectSlug: PropTypes.string.isRequired,
+      }).isRequired,
+    };
+
+    trackRedirect = nextRoute => {
+      const {pathname, search} = this.props.location;
+
+      const payload = {
+        from: `${pathname}${search}`,
```

Comment:
Using the pathname will be hard to run queries on because the URLs are dynamic because of project/org. I would normalize the route path with `getRouteStringFromRoutes` instead.

We can get the org id (the id not slug) from organization from context (it should be in context, I haven't looked at the routes, but these routes should be nested under `OrganizationDetails`, if not I think just having the slug is fine).